### PR TITLE
Fix database methods are ambiguous without curlies

### LINKED ISSUE #201: Database methods are ambiguous without curlies
When using `database.read/write` or `database.read/writeTransaction` and using it with a one-liner lambda without curly braces, the call is **ambiguous**.

ambiguous example

We either have to explicitly cast or use curlies. But when adding curlies, we get a linter warning about having unecessary curlies. This ends up in polluting the code base with annoying warning suppressions `@SuppressWarnings("java:S1602")`, since this happens pretty much in majority of all database use cases.

The code base already has some of those suppressions in it and it keeps getting more.

suppression

----

A fix for this could be to rename the methods so that they are not overloads anymore. It is super unfortunate, but I think it would be better than having to go through this cumbersome annoyance each time using the database class.

### PR DESCRIPTION
**Pull request by Zabuzard**
### Overview

Fixes #201 .

**tl;dr,** the database interface lead to annoying situations, such as:

Its now fixed by renaming the methods slightly:
* the `void read` method got renamed to `readAndConsume`
* the non-void `T write` method got renamed to `writeAndProvide`

(analogously for the transaction methods)

This also fixes all currently existing instances of `java:S1602`, which have been **suppressed** so far.

### Details

I did not rename either both void or both non-void methods on purpose. Mostly to reduce the otherwise heavy merge conflicts. The `void read` method is the one that is generally not used, same for the `T write` method. Hence I only renamed the "rarely used" methods. Context-wise it also makes sense that `read(...)` actually returns a result while `write(...)` usually does not return a result.

### Merge conflicts

I made the changes in a way that the merge conflicts are **as minimal as possible**. In general, you should be able to just accept any change without any interferences (unless you changed the logic itself that this refactoring touches - unlikely though).

### CODE REVIEW

**Review by RealYusufIsmail — APPROVED** [review]

**Review by java-coding-prodigy — APPROVED** [review]
All the unit tests pass, and you must have tested this manually too, LGTM

**Review by Tais993 — APPROVED** [review]
